krest.py

```diff
@@ -60,21 +74,35 @@ class RetryCfg(object):
         on_other_errors = True
         on_toofast_error = False
 
-    def __init__(self, k2_addr, username, password, ssl_validate=True,
+    def __init__(self, k2_addr, auth: AuthBase=None, username=None, password=None, sdp_id=None,
```

Comment:
To maintain backwards compatibility, we should keep the username and password parameters in the same place:
```suggestion
    def __init__(self, k2_addr, username=None, password=None, auth: AuthBase = None, sdp_id=None,
```